petastorm/pytorch.py

```diff
@@ -129,7 +129,8 @@ def __init__(self, reader, batch_size=1, collate_fn=decimal_friendly_collate,
         if shuffling_queue_capacity > 0:
             # We can not know what is the reasonable number to use for the extra capacity, so we set a huge number
             # and give up on the unbound growth protection mechanism.
-            min_after_dequeue = shuffling_queue_capacity - 1
+            if min_after_dequeue == 0:
```

Comment:
I'm afraid that I'm not clear what you mean. Should I just default min_after_dequeue to 0 like tf_tensors()?
I'm probably brainwashed from working with .net for so long, but I am usually hesitant to change interfaces and thus require changes from callers. My goal was to allow callers to provide the parameter if they want to, but otherwise maintain the current behavior. Would using a default value of -1 to recognize when to use the originally hard-coded value of (shuffling_queue_capacity - 1) be reasonable?

Reply:
Thinking about it more, you are probably right. We can keep the default as `shuffling_queue_capacity-1`.

> Would using a default value of -1 to recognize when to use the originally hard-coded value of (shuffling_queue_capacity - 1) be reasonable

How about `min_after_dequeue=None`? What I mean is that `0` in `min_after_dequeue=0` can actually make some limited sense as in `RandomShufflingBuffer(capacity, min_after_retrieve=0)` - we deplete the entire buffer before re-accumulating and reshuffling it.